csharp/private/actions/assembly.bzl

```diff
@@ -52,9 +52,10 @@ def AssemblyAction(
         toolchain):
     out_ext = "dll" if target == "library" else "exe"
 
-    out = actions.declare_file("%s.%s" % (name, out_ext))
-    refout = actions.declare_file("%s.ref.%s" % (name, out_ext))
-    pdb = actions.declare_file(name + ".pdb")
+    sanitized_tfm = target_framework.replace(".", "_")
+    out = actions.declare_file("%s.%s.%s" % (name, sanitized_tfm, out_ext))
```

Comment:
Don't suppose it's possible to put them in separate directories (like is the default behaviour for msbuild/csproj)

Reply:
I don't think so. It definitely can't cross package boundaries which would make it real awkward.

Reply:
https://docs.bazel.build/versions/master/skylark/lib/actions.html#declare_file seems to indicate it could be. I'll try making the path be the TFM

Reply:
```
$ bazel build :Top
INFO: Analyzed target //examples/diamond:Top (13 packages loaded, 69 targets configured).
INFO: Found 1 target...
Target //examples/diamond:Top up-to-date:
  bazel-bin/examples/diamond/net48.bazelout/Top.dll
  bazel-bin/examples/diamond/net48.bazelout/Top.ref.dll
  bazel-bin/examples/diamond/net48.bazelout/Top.pdb
INFO: Elapsed time: 4.276s, Critical Path: 3.87s
INFO: 6 processes: 6 darwin-sandbox.
INFO: Build completed successfully, 7 total actions
```

Thanks. That's a better outcome. I added `.bazelout` to at make it less likely to collide with a "real" directory.